math/src/elliptic_curve/short_weierstrass/curves/bls12_381/compression.rs

```diff
@@ -0,0 +1,172 @@
+use super::field_extension::BLS12381PrimeField;
+use crate::cyclic_group::IsGroup;
+use crate::elliptic_curve::short_weierstrass::point::ShortWeierstrassProjectivePoint;
+use crate::elliptic_curve::traits::FromAffine;
+use crate::field::element::FieldElement;
+use crate::unsigned_integer::element::U256;
+use crate::{
+    elliptic_curve::short_weierstrass::curves::bls12_381::curve::BLS12381Curve,
+    errors::ByteConversionError, traits::ByteConversion,
+};
+use std::cmp::Ordering;
+use std::ops::Neg;
+
+pub type G1Point = ShortWeierstrassProjectivePoint<BLS12381Curve>;
+pub type BLS12381FieldElement = FieldElement<BLS12381PrimeField>;
+const MODULUS: U256 =
+    U256::from("73eda753299d7d483339d80809a1d80553bda402fffe5bfeffffffff00000001");
+
+pub fn check_point_is_in_subgroup(point: &G1Point) -> bool {
+    let inf = G1Point::neutral_element();
+    let aux_point = point.operate_with_self(MODULUS);
+    inf == aux_point
+}
+
+pub fn decompress_g1_point(input_bytes: &mut [u8; 48]) -> Result<G1Point, ByteConversionError> {
+    let first_byte = input_bytes.first().unwrap();
+    // We get the first 3 bits
+    let prefix_bits = first_byte >> 5;
+    let first_bit = (prefix_bits & 4_u8) >> 2;
+    // If first bit is not 1, then the value is not compressed.
+    if first_bit != 1 {
+        return Err(ByteConversionError::ValueNotCompressed);
+    }
+    let second_bit = (prefix_bits & 2_u8) >> 1;
+    // If the second bit is 1, then the compressed point is the
+    // point at infinity and we return it directly.
+    if second_bit == 1 {
+        return Ok(G1Point::neutral_element());
+    }
+    let third_bit = prefix_bits & 1_u8;
+
+    let first_byte_without_contorl_bits = (first_byte << 3) >> 3;
+    input_bytes[0] = first_byte_without_contorl_bits;
```

Comment:
I think var `first_byte_without_contorl_bits` should be `first_byte_without_control_bits`